digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala

```diff
@@ -747,6 +802,62 @@ trait LaneOperations {
     }
   }
 
+  def processLanesByRoadAddress(newLanes: Set[NewLane], laneRoadAddressInfo: LaneRoadAddressInfo,
+                                   username: String): Set[Long] = {
+    withDynTransaction {
+      val filteredRoadAddresses = LaneUtils.getRoadAddressToProcess(laneRoadAddressInfo)
+      //Get only the lanes to create
+      val lanesToInsert = newLanes.filter(_.id == 0)
+      val clickedMainLane = newLanes.filter(_.id != 0).head
+      val selectedLane = getPersistedLanesByIds(Set(clickedMainLane.id), newTransaction = false).head
+
+      val roadLinkIds = filteredRoadAddresses.map(_.linkId)
+      val linksWithSideCodes = getLinksWithCorrectSideCodes(selectedLane, roadLinkIds, newTransaction = false)
+
+      // Throw error if links are not consecutive
+      if (linksWithSideCodes.size != roadLinkIds.size)
+        throw new InvalidParameterException(s"All links in selection does not have road address.")
+
+      val existingLanes = fetchAllLanesByLinkIds(roadLinkIds.toSeq, newTransaction = false)
```

Comment:
onko tarvetta tarkistaa onko aluella kaistaa, eikös tällä lisätä ensimmäinen lisäkaista jonka jälkeen ei voi lisätä toista lisäkaistaa tieosoitteen avulla?

Reply:
UI:ssa pystyy antamaan kaistan alku- ja loppupisteiksi minkä tahansa tieosan ja etäisyyden, joten käytännössä voidaan aloittaa tai lopettaa kaista sellaiseen kohtaan missä on jo lisäkaista samalla koodilla. Jonkinlainen tarkastus pitänee olla näitä tilanteita varten

Reply:
ja tulee kysymys tehdäänkö UI puolella vai backend puolella. Miten käyttäjä saa tiedon että väli on liian pitkä? Voisko tarkistuksen siirtää UI ja sitten Backend olettaisi aluen olevan aina validi. Käyttäjälle tulisi viesti että alue on liian pitkä.

Reply:
Mitäs product owner tuumisi?

Reply:
varmaan validointi pitää tehdä kummassakin puolessa jotta ollaan varmoja että virhettä ei tapahdu